### PR TITLE
Migrate dict.c unit tests to new framework

### PR DESCRIPTION
**Pull request by haoqixu**
This PR migrates the tests related to dict into new test framework as part of #428.

### CODE REVIEW

File: src/unit/test_dict.c [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    monotonicInit(); /* Required for dict tests, that are relying on monotime during dict rehashing. */
+
+    return 0;
+}
+
+
```
Comment (enjoy-binbin) [inline src/unit/test_dict.c]: I see there are two line breaks in some places here, can you remove it?

Comment (haoqixu) [reply src/unit/test_dict.c]: done

**Review by enjoy-binbin — APPROVED** [review]
seem ok to me.

File: src/unit/test_dict.c [hunk 2]
```diff
@@ -0,0 +1,341 @@
+#include "../dict.c"
+#include "test_help.h"
+
+#define TEST(name) printf("test — %s\n", name);
```
Comment (hpatro) [inline src/unit/test_dict.c]: I guess we don't need this macro anymore.

File: src/unit/test_dict.c [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+int test_dictRelease(int argc, char **argv, int flags) {
+    /* Restore to original state */
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    dictRelease(_dict);
+
+    return 0;
+}
```
Comment (hpatro) [inline src/unit/test_dict.c]: This test doesn't have any meaning. Could you reset the dictionary in the previous/next test?

**Review by hpatro — COMMENTED** [review]
Thanks for working on it!

File: src/unit/test_dict.c (hunk 3 above)
Comment (haoqixu) [reply src/unit/test_dict.c]: done

**Comment by zuiderkwast** [comment]
@haoqixu The FORTIFY_SOURCE problem is solved on the unstable branch. Can you merge latest unstable to this branch?

**Comment by enjoy-binbin** [comment]
Merged, thanks for the work.

**Pull request merged**